R/epi_workflow.R

```diff
@@ -58,6 +58,103 @@ is_epi_workflow <- function(x) {
   inherits(x, "epi_workflow")
 }
 
+
+#' Add a model to an `epi_workflow`
+#'
+#' @seealso [workflows::add_model()]
+#' - `add_model()` adds a parsnip model to the `epi_workflow`.
+#'
+#' - `remove_model()` removes the model specification as well as any fitted
+#'   model object. Any extra formulas are also removed.
+#'
+#' - `update_model()` first removes the model then adds the new
+#' specification to the workflow.
+#'
+#' @details
+#' Has the same behaviour as [workflows::add_model()] but also ensures
+#' that the returned object is an `epi_workflow`.
+#'
+#' @inheritParams workflows::add_model
+#'
+#' @param x An `epi_workflow`.
+#'
+#' @param spec A parsnip model specification.
+#'
+#' @param ... Not used.
+#'
+#' @return
+#' `x`, updated with a new, updated, or removed model.
+#'
+#' @export
+#' @examples
+#' jhu <- case_death_rate_subset %>%
+#' dplyr::filter(time_value > "2021-11-01",
+#' geo_value %in% c("ak", "ca", "ny"))
+#'
+#' r <- epi_recipe(jhu) %>%
+#'   step_epi_lag(death_rate, lag = c(0, 7, 14)) %>%
+#'   step_epi_ahead(death_rate, ahead = 7)
+#'
+#' rf_model <- rand_forest(mode = "regression")
+#'
+#' wf <- epi_workflow(r)
+#'
+#' wf <- wf %>% add_model(rf_model)
+#' wf
+#'
+#' lm_model <- parsnip::linear_reg()
+#'
+#' wf <- update_model(wf, lm_model)
+#' wf
+#'
+#' wf <- remove_model(wf)
+#' wf
+#' @export
+add_model <- function(x, spec, ..., formula = NULL)
+  UseMethod('add_model')
+
+#' @rdname add_model
+#' @export
+remove_model <- function(x)
+  UseMethod('remove_model')
+
+#' @rdname add_model
+#' @export
+update_model <- function(x, spec, ..., formula = NULL)
+  UseMethod('update_model')
```

Comment:
Please run `styler::style_pkg()`. It will handle the missing `{` and `}` for these.